### PR TITLE
[Contract][Relay] fix: Update IPFS content handling (from content to cid)

### LINKED ISSUE #170: [Contract] [Relay] Consistent on the CID storing pattern
**Describe the bug**

Improve consistency on the CID storage pattern by storing CID in the contract.

**Expected behavior**

- [Store CID in the contract](https://github.com/social-tw/social-tw-website/blob/c1431ae1ba1599a77f19403ba34c1b31d1e4226b/packages/contracts/contracts/UnirepApp.sol#L84)

 - [Send CID as part of the encoded data](https://github.com/social-tw/social-tw-website/blob/c1431ae1ba1599a77f19403ba34c1b31d1e4226b/packages/relay/src/routes/post.ts#L120)

### PR DESCRIPTION
**Pull request by FoodChain1028**
## Summary

Storing cid into `Post` and `LeaveComment` on chain.

## Linked Issue

close #170

## Details

See relay PostService and CommentService

## Impacted Areas

Would change the `content` storage on chain.

## Tests

- [ ] check if comment is cid in synchroniser
- [ ] check if post is cid in synchroniser 

## Verification Steps

`yarn relay test`

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [ ] Draft implementation for `function`
-   [ ] Unit tests for `function`

## Checklist

-   [ ] All new and existing tests pass
-   [ ] I have updated the documentation (if applicable)
-   [ ] My changes do not introduce new security vulnerabilities

### CODE REVIEW

File: packages/relay/test/comment.test.ts [hunk 1]
```diff
@@ -107,7 +115,32 @@ describe('COMMENT /comment', function () {
...
+        const unirepAppInterface = new ethers.utils.Interface(abi)
+        const rawEvent = receipt.logs
+            .map((log) => unirepAppInterface.parseLog(log))
+            .find((log) => log.name == 'Comment')
+        const commentEvent = rawEvent?.args
+
+        expect(commentEvent).to.not.be.undefined
+        if (commentEvent) {
+            expect(commentEvent[0].toString()).equal(epk.toString())
+            expect(commentEvent[1]).equal('0')
```
Comment (ryanycw) [inline packages/relay/test/comment.test.ts:138]: Recommend to fetch latestPostId before creating post and use that as the postId, and same for the other emit variables. Hard code will make it harder if we want to adjust the test or test tools in the future.

File: packages/relay/test/synchronizer.test.ts [hunk 2]
```diff
@@ -10,6 +10,109 @@ import { signUp } from './utils/signUp'
...
+    let sync: UnirepSocialSynchronizer
+    let unirep: Unirep
+    let unirepApp: UnirepApp
+    let users: {
+        hashUserId: String
+        wallet: any
+        userState: UserState
+    }[] = []
+
+    before(async function () {
```
Comment (ryanycw) [inline packages/relay/test/synchronizer.test.ts:27]: Seems like most of the test setup is the same between Post and Comment. We can consider to separate it into another utils file for this.

**Review by ryanycw — APPROVED** [review]
AWESOME, minot comments only.

File: packages/relay/test/synchronizer.test.ts (hunk 2 above)
Comment (FoodChain1028) [reply packages/relay/test/synchronizer.test.ts:27]: I will leave this currently and merge first.

**Pull request merged**